### PR TITLE
Add support for X708 UPS HAT

### PR DESCRIPTION
**Pull request by HiLivin**
This adds a possibility to install X708 scripts with `blitz.ups.sh on x708`.

Succesfully tested the code on the `dev` branch with Raspberry Pi 4B+ and  X708 v1.2 UPS HAT.

- [x] The [`blitz.shutdown.sh`](https://github.com/rootzoll/raspiblitz/blob/v1.7/home.admin/config.scripts/blitz.shutdown.sh) is executed before cutting off the power to the Raspberry Pi

- [x] The I2C is being enabled to allow getting data from the on-board battery management chip. I haven't found any issues as referenced by #1058 

- [x] The battery voltage and AC state are correctly fetched by [`_background.scan.sh`](https://github.com/rootzoll/raspiblitz/blob/v1.7/home.admin/_background.scan.sh) and later displayed on the status window

- [x] Still, I'd like to know if it's alright to link to my repo for downloading the scripts or maybe it would be better to pack them and put somewhere along with other RaspiBlitz code.

### CODE REVIEW

**Comment by rootzoll** [comment]
Thats an interesting integration. I see if I can get a hands on those boards to test your integration. I added it to the v1.8.2 milestone for possible merge .. a bit up the line because v1.8.0 & 1.8.1 wil be very WebUI focused.

> Still, I'd like to know if it's alright to link to my repo for downloading the scripts or maybe it would be better to pack them and put somewhere along with other RaspiBlitz code.

You can keep things in your repo and install it on demand, but then after cloning the repo within RaspiBlitz script it should checkout to a certain commit hash for security reasons.

**Comment by HiLivin** [comment]
It would be great to have someone double-check the whole thing. 

- [x] I've added the `git checkout` with a proper hash to the install script and squashed the commits

**Comment by rootzoll** [comment]
Progress looks great. If you think you have everything in there you can remove `[WIP]` from title and mark it with the tag `final testing`. Targeting here for release with 1.8.1

**Comment by HiLivin** [comment]
Done changing title. I think only repo maintainers can add tags/labels.

**Comment by HiLivin** [comment]
Bumped the repo version to checkout. A major improvement was added.

> Normally the x708 board waits only for ~50 seconds for the Pi to shutdown. If Pi shutdowns later than that, then the UPS hat doesn't turn itself off and the power to the Pi isn't cut. In case of power loss it wouldn't be a much problem - the board would eventually turn off because of low battery.

Now, to do it right, when the x708 starts waiting for Pi to shutdown a background loop is started with a specific timeout after which the UPS is triggered to wait for Pi shutdown again. This way the RaspiBlitz shutdown script can take very long and the hat will properly cut off the power afterwards.

**Comment by HiLivin** [comment]
Bumped again.
The install script now uses `patch` on `00mainMenu.sh` instead of changing code lines with `sed` - this will improve compatibility with future versions of menu script.

The functionality seems fully developed. It would be nice to get some feedback from others using this UPS board with the scripts.

**Comment by rootzoll** [comment]
@HiLivin if you prepare a tweet asking for broader feedback on this feature (Call for Testers) .. I am happy to share.

**Comment by holgern** [comment]
I have installed the X708 V2.0 and I have applied this PR to 1.7.2. After reboot, it is doing now every time a shutdown.
I have two 18650  (connected in parallel with the holder) and I power the x708 board through USB.

I will wait what will happen when the battery are fully loaded. (They are now at 25%)

**Comment by holgern** [comment]
After reaching MIN_VOLTAGE = 4.0, my raspi could start.
sudo ./config.scripts/blitz.ups.sh status returns now:
upsStatus='ONLINE'
upsBattery='4.22V'

**Comment by HiLivin** [comment]
Great! So that's a new board.

The automatic shutdown should be called when `battery voltage is low` AND `AC power loss is detected` (PLD pin high). 
There could possibly be some hardware issues with PLD when batteries are low and the power supply is inefficient - high voltage drop might be recognized as AC power fail.

File: home.admin/config.scripts/blitz.ups.sh [hunk 1]
```diff
@@ -48,13 +49,37 @@ if [ "$1" = "1" ] || [ "$1" = "on" ]; then
...
+    # enable I2C interface
+    sudo raspi-config nonint do_i2c 0
+
+    # install prerequisites
+    sudo apt-get install -y python3-smbus i2c-tools
+
+    # clone git repository with the X708 scripts
+    git clone https://github.com/HiLivin/x708blitz.git /home/admin/x708blitz
+    cd /home/admin/x708blitz
+    git checkout ed3612617365ba485141b98c595bdba67be95804
```
Comment (holgern) [inline home.admin/config.scripts/blitz.ups.sh]: This needs to change to the newest commit
```
git checkout 2ed6caffedef3bd523732287234ebe9d200fd92f
```

**Comment by YBHQ** [comment]
#

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: home.admin/config.scripts/blitz.ups.sh (hunk 1 above)
Comment (HiLivin) [reply home.admin/config.scripts/blitz.ups.sh]: Done in 7782c1935f06c48ba6feb76e048547ed2d70bc60.

**Comment by YBHQ** [comment]
~~~~

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by smamamo** [comment]
I have installed the X708 V2.0 on my raspiblitz V. 018.0 and all works . 
Auto start - Start when power cam back works as well 
many thanks

**Comment by rootzoll** [comment]
Perfect - so I am merging to be included in v1.8.1 release.

**Comment by smamamo** [comment]
A suggestion would be the integration of a simple monitoring of the battery charge. - x708.info.py 
And a simple adjustment of the fan speed. - x708.fan.py -

My battery setting

MIN_VOLTAGE = 3.7 # (volts) shutdown when voltage drops below this value

I optimised my fans at

HIGH_THRESHOLD = 56 # (degrees Celsius) Fan running at high speed at this temperature.
LOW_THRESHOLD = 50 # (degrees Celsius) Fan running at low speed at this temperature.
SLEEP_INTERVAL = 5 # (seconds) How often do we check the core temperature.

**Comment by rootzoll** [comment]
@smamamo if you like to add furthe rimprovements, it best to start a new PR on it

**Pull request merged**